packages/backend/src/models/SavedChartModel.ts

```diff
@@ -416,24 +403,31 @@ export class SavedChartModel {
     async getLatestVersionSummaries(
         chartUuid: string,
     ): Promise<ChartVersionSummary[]> {
-        const getLastVersionUuidSubQuery =
-            this.getLastVersionUuidQuery(chartUuid);
         const { daysLimit } = this.lightdashConfig.chart.versionHistory;
         const chartVersions = await this.getVersionSummaryQuery()
             .where(`${SavedChartsTableName}.saved_query_uuid`, chartUuid)
-            .andWhere(function whereRecentVersionsOrCurrentVersion() {
-                // get all versions from the last X days + the current version ( in case is older than X days )
-                this.whereRaw(
-                    `${SavedChartVersionsTableName}.created_at >= DATE(current_timestamp - interval '?? days')`,
-                    [daysLimit],
-                ).orWhere(
-                    `${SavedChartVersionsTableName}.saved_queries_version_uuid`,
-                    getLastVersionUuidSubQuery,
-                );
-            })
             .orderBy(`${SavedChartVersionsTableName}.created_at`, 'asc');
 
-        return chartVersions.map(SavedChartModel.convertVersionSummary);
+        return chartVersions
+            .reduce<VersionSummaryRow[]>((acc, version) => {
```

Comment:
I'm concerned about performance as we will fetch all the versions from the warehouse. 
What if we keep the existing query and add `if(rows.length < 2) { ... run second query to fetch the last version }`

Reply:
I though performance was not going to be an issue because we purge versions periodically (do we ?) 

But yeah, we can do another query

Reply:
We don't purge versions yet.

Reply:
Updated to make  a separate query, also updated tests